linera-chain/src/lib.rs

```diff
@@ -126,10 +126,11 @@ pub enum ChainError {
     OwnerWeightError(#[from] WeightedError),
 }
 
-#[derive(Debug)]
+#[derive(Copy, Clone, Debug)]
 pub enum ChainExecutionContext {
     Query,
     DescribeApplication,
     IncomingMessage(u32),
     Operation(u32),
+    Certificate,
```

Comment:
`Block` is more precise because it will happen when the block is proposed.

Reply:
Ok.